File: DotNet/Models/VehicleDetail.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Vehicle_Configurator_Project.Models
{
    [Table("vehicle_detail")]
    public class VehicleDetail
    {
        [Key]
        [Column("config_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ConfigId { get; set; }

        [ForeignKey(nameof(Model))]
        [Column("model_id")]
        [JsonIgnore] // Prevent cyclic reference when serializing
        public int ModelId { get; set; }

        [JsonIgnore]
        public Model Model { get; set; }

        [ForeignKey(nameof(Component))]
        [Column("comp_id")]
        public int? ComponentId { get; set; }
        public Component Component { get; set; }

        [Column("comp_type")]
        [Required]
        public char CompType { get; set; }

        [Column("is_configurable")]
        [Required]
        public char IsConfigurable { get; set; }
    }
}

File: .NET & JAVA/JAVA/VehicleConfigurator/src/main/java/com/example/entities/VehicleDetail.java
package com.example.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

@Entity
@Table(name = "vehicle_detail")
public class VehicleDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "config_id")
    private int configId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "model_id", nullable = false)
    @JsonIgnore // Prevent cyclic reference when serializing
    private Model model;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "comp_id", referencedColumnName = "comp_id")
    private Component component;

    @Column(name = "comp_type", nullable = false)
    private char compType;

    @Column(name = "is_configurable", nullable = false)
    private char isConfigurable;

    // Constructors
    public VehicleDetail() {}

    // Getters and Setters
    public int getConfigId() {
        return configId;
    }

    public void setConfigId(int configId) {
        this.configId = configId;
    }

    public Model getModel() {
        return model;
    }

    public void setModel(Model model) {
        this.model = model;
    }

    public Component getComponent() {
        return component;
    }

    public void setComponent(Component component) {
        this.component = component;
    }

    public char getCompType() {
        return compType;
    }

    public void setCompType(char compType) {
        this.compType = compType;
    }

    public char getIsConfigurable() {
        return isConfigurable;
    }

    public void setIsConfigurable(char isConfigurable) {
        this.isConfigurable = isConfigurable;
    }
}